prometheus/src/bank_metrics.rs

```diff
@@ -15,7 +15,8 @@ pub fn write_bank_metrics<W: io::Write>(
             help: "Block Slot",
             type_: "gauge",
             metrics: banks_with_commitments
-                .for_each_commitment(|bank| Metric::new(bank.clock().slot)),
+                .for_each_commitment(|bank| Some(Metric::new(bank.clock().slot)))
+                .unwrap(),
```

Comment:
Why does `for_each_commitment` now return something that needs to be unwrapped? And why is that safe to do here? Can’t we make it always succeed? If all calls to the closure return `None`, we would just have no metrics. That’s fine. We'd print a `# HELP` and `# TYPE` line for something that is missing then, but then a human can go and look and hopefully figure out why it’s missing.